packages/core/src/ProxyConfiguration.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * Proxy server configuration.
+ */
+class ProxyConfiguration {
+    constructor(
+        /**
+         * Proxy type.
+         */
+        readonly type: ProxyType,
+        /**
+         * Proxy address. Can be either in the IP address format, ex. "1.1.1.1", or hostname, ex. "example.com".
+         */
+        readonly address: string,
+        /**
+         * Proxy port.
+         */
+        readonly port: number,
+        /**
+         * Username for Basic authentication scheme.
+         *
+         * Note: in case of Android, credentials are working out of the box only for "http" or "https" proxy types.
+         * In case of "socks" type you need to create a native module which will set "java.net.Authenticator", because
+         * OkHttp library doesn't support SOCKS+authentication proxy scheme.
+         */
+        readonly username?: string,
+        /**
+         * Password for Basic authentication scheme.
+         */
+        readonly password?: string
+    ) { }
+}
+
+enum ProxyType {
+    HTTP = "http",
```

Comment:
We probably should not allow HTTP proxy, since most of the device will prevent clear text request anyway. Nowadays HTTPS is the recommended standard

Reply:
`HTTP` proxy doesn't mean data will be transferred in clear-text. If connection to the target server is done via `HTTPS` protocol, then in case of `HTTP` so called tunnel will be opened (`TCP Connect`) to the target server, eavesdropping is not possible.